src/txn/txn.c

```diff
@@ -1402,8 +1419,8 @@ __wt_txn_init(WT_SESSION_IMPL *session, WT_SESSION_IMPL *session_ret)
 
 #ifdef HAVE_DIAGNOSTIC
     if (S2C(session_ret)->txn_global.states != NULL) {
-        WT_TXN_STATE *txn_state;
-        txn_state = WT_SESSION_TXN_STATE(session_ret);
+        WT_TXN_SHARED *txn_state;
+        txn_state = WT_SESSION_TXN_SHARED(session_ret);
```

Comment:
If we write this as:
```
WT_ASSERT(session, S2C(session_ret)->txn_global.states == NULL ||
      WT_SESSION_TXN_SHARED(session_ret)->pinned_id == WT_TXN_NONE);
```
I don't think we need the #ifdef or local variable.

Reply:
c12daa2